runtime/src/main/cpp/Memory.cpp

```diff
@@ -2022,6 +2121,7 @@ OBJ_GETTER(allocInstance, const TypeInfo* type_info) {
 #endif  // USE_GC
   auto container = ObjectContainer(state, type_info);
   ObjHeader* obj = container.GetPlace();
+  CycleDetector::insertCandidateIfNeeded(obj);
```

Comment:
The meaning is quite similar to what `cyclicAddAtomicRoot` does, so it might be better to place `insertCandidateIfNeeded` near `cyclicAddAtomicRoot`.
(For example, `cyclicAddAtomicRoot` must be after `rememberNewContainer` to prevent data race, the same might be required for `insertCandidateIfNeeded` too).